src/storage/mvcc/reader/scanner/forward.rs

```diff
@@ -598,10 +601,44 @@ impl<S: Snapshot> ScanPolicy<S> for DeltaEntryPolicy {
             } else {
                 Ok((vec![], vec![]))
             };
+            let mut old_value = None;
+            if check_lock_extra_op(self.extra_op, lock.lock_type) {
+                while Key::is_user_key_eq(
+                    cursors.write.key(&mut statistics.write),
+                    current_user_key.as_encoded(),
+                ) {
+                    assert_ge!(
+                        lock.ts,
+                        Key::decode_ts_from(cursors.write.key(&mut statistics.write))?
+                    );
```

Comment:
I think it's not always true when there are pessimistic transactions.

Reply:
Here we have already skipped pessimistic txns.